tests/cpp_unit_tests/test_deserializer.cpp

```diff
@@ -279,14 +279,34 @@ TEST_CASE("Deserializer") {
 
         SUBCASE("Check buffer") {
             auto const& info = deserializer.get_dataset_info();
-            CHECK(info.get_component_info("node").elements_per_scenario == 3);
-            CHECK(info.get_component_info("node").total_elements == 3);
-            CHECK(info.get_component_info("line").elements_per_scenario == 2);
-            CHECK(info.get_component_info("line").total_elements == 2);
-            CHECK(info.get_component_info("source").elements_per_scenario == 3);
-            CHECK(info.get_component_info("source").total_elements == 3);
-            CHECK(info.get_component_info("sym_load").elements_per_scenario == 4);
-            CHECK(info.get_component_info("sym_load").total_elements == 4);
+
+            auto const& node_info = info.get_component_info("node");
+            CHECK(node_info.elements_per_scenario == 3);
+            CHECK(node_info.total_elements == 3);
+            CHECK(node_info.has_attribute_indications == true);
+            CHECK(node_info.attribute_indications[0]->name == "id"s);
+            CHECK(node_info.attribute_indications[1]->name == "u_rated"s);
```

Comment:
please use `REQUIRE` to check before doing unchecked element access (`REQUIRE` is fatal while `CHECK` is not)
```suggestion
            CHECK(node_info.has_attribute_indications == true);
            REQUIRE(node_info.attribute_indications.size() == 2);
            CHECK(node_info.attribute_indications[0]->name == "id"s);
            CHECK(node_info.attribute_indications[1]->name == "u_rated"s);
```